EdFi.Ods.Utilities.Migration/logistics/scripts/modules/path-resolver.psm1

```diff
@@ -0,0 +1,153 @@
+    #Remove Case Sensitivity
+    $path = $path.ToLower()
+    $itemName = $itemName.ToLower()
+
+    $pos = $path.LastIndexOf("\$itemName\")
+
+    if ($pos -lt 0) {
+        if ($path.EndsWith("\$itemName")) {
+            $pos = $path.LastIndexOf("\$itemName")
+        }
+        else {
+            throw "Unable to locate '$itemName' in path '$path'."
+        }
+    }
+
+    $newPath = $path.Substring(0, $pos) + "\$itemName"
+    return (Resolve-Path $newPath)
+}
+
+function Get-RootPath {
+    #Look up from this script's location to find the highest level common named folder.
+    $logisticsBasePath = Get-AncestorItemPath $PSScriptRoot "logistics"
+    #Jump up two levels to get the root.
+    return Resolve-Path "$logisticsBasePath\..\..\"
+}
+
+#Using the Root path for the repositories, return back the path for the $repositoryName specified.
+function Get-RootBasedRepositoryPath ([string]$repositoryName) {
+    $root = Get-RootPath
+    $repositoryPath = "$($root)$repositoryName"
+
+    If (-not (Test-Path $repositoryPath)) {
+        throw ("$repositoryName path not found under $($root)!")
+    }
+    return $repositoryPath
+}
+
+<#
+.Synopsis
+Return paths to specified files, checking all repositories, and including only the most specific version of each file.
+
+.Description
+Return paths to specified files, checking all repositories, and including only the most specific version of each file.
+
+For example, in a system with three repositories, laid out like this:
+
+    \Ed-Fi-Core
+        \1234.txt
+        \qwer.txt
+    \Ed-Fi-Plugins
+        \qwer.txt
+        \asdf.txt
+    \Ed-Fi-Apps
+        \asdf.txt
+        \zxcv.txt
+
+Seraching for zxcv.txt or asdf.txt returns the version in Ed-Fi-Apps; searching for qwer.txt returns the version in Ed-Fi-Plugins, and searching for 1234.txt returns the version in Ed-Fi-Core.
+
+.Parameter PathSuffix
+The path suffix to append to the repository path. E.g. if $pathSuffix is 'logistics', find files in <most specific repository>\logistics.
+#>
+function Get-RepositoryResolvedPath ([string]$pathSuffix) {
+    foreach ($repositoryName in $invertedRepositoryNames) {
+        $repositoryPath = Get-RootBasedRepositoryPath $repositoryName
+        If (Test-Path "$repositoryPath\$pathSuffix") {
+            return Resolve-Path "$repositoryPath\$pathSuffix"
+        }
+    }
+    $errorMsg = "$pathSuffix was not found in the following repositories:`r`n{0}" -f ($invertedRepositoryNames -join "`r`n")
+    throw ($errorMsg)
+}
+
+
+#Sets a global folders variable. Is the primary access mechanism for resolving paths. Utilizes a delegate.
+if (-not ($script:folders)) {
+    $script:folders = @{ }
+    #This is used in remote web deployments to find this path resolver and CANNOT be a delegate.
+    $folders.core = Get-CorePath
+    $folders.scripts = [System.Func[Object, Object]] { return (Get-RepositoryResolvedPath "logistics\scripts\$($args[0])") }
```

Comment:
if we keep path resolving, then lets remove the logistics in the folder structure

Reply:
I remove it . Thanks